image_generation/lcm_dreamshaper_v7/cpp/src/main.cpp

```diff
@@ -0,0 +1,289 @@
+#include "openvino_extensions/strings.hpp"
+
+#include "cxxopts.hpp"
+#include "scheduler_lcm.hpp"
+#include "lora_cpp.hpp"
+#include "imwrite.hpp"
+
+ov::Tensor randn_tensor(uint32_t height, uint32_t width, bool use_np_latents, uint32_t seed = 42) {
+    ov::Tensor noise(ov::element::f32, {1, 4, height / 8, width / 8});
+    if (use_np_latents) {
+        // read np generated latents with defaut seed 42
+        const char * latent_file_name = "../scripts/np_latents_512x512.txt";
+        std::ifstream latent_copy_file(latent_file_name, std::ios::ate);
+        OPENVINO_ASSERT(latent_copy_file.is_open(), "Cannot open ", latent_file_name);
+
+        size_t file_size = latent_copy_file.tellg() / sizeof(float);
+        OPENVINO_ASSERT(file_size >= noise.get_size(), "Cannot generate ", noise.get_shape(), " with ", latent_file_name, ". File size is small");
+
+        latent_copy_file.seekg(0, std::ios::beg);
+        for (size_t i = 0; i < noise.get_size(); ++i)
+            latent_copy_file >> noise.data<float>()[i];
+    } else {
+        std::mt19937 gen{static_cast<unsigned long>(seed)};
+        std::normal_distribution<float> normal{0.0f, 1.0f};
+        std::generate_n(noise.data<float>(), noise.get_size(), [&]() {
+            return normal(gen);
+        });
+    }
+    return noise;
+}
+
+struct StableDiffusionModels {
+    ov::CompiledModel text_encoder;
+    ov::CompiledModel unet;
+    ov::CompiledModel vae_decoder;
+    ov::CompiledModel tokenizer;
+};
+
+StableDiffusionModels compile_models(const std::string& model_path, const std::string& device,
+                                     const std::string& type, const std::string& lora_path,
+                                     const float alpha, const bool use_cache) {
+    StableDiffusionModels models;
+
+    ov::Core core;
+    if (use_cache)
+        core.set_property(ov::cache_dir("./cache_dir"));
+    core.add_extension(TOKENIZERS_LIBRARY_PATH);
+
+    std::map<std::string, float> lora_models;
+    lora_models[lora_path] = alpha;
+
+    // CLIP and UNet
+    auto text_encoder_model = core.read_model((model_path + "/" + type + "/text_encoder/openvino_model.xml").c_str());
+    auto unet_model = core.read_model((model_path + "/" + type + "/unet/openvino_model.xml").c_str());
+    auto text_encoder_and_unet = load_lora_weights_cpp(core, text_encoder_model, unet_model, device, lora_models);
+    models.text_encoder = text_encoder_and_unet[0];
+    models.unet = text_encoder_and_unet[1];
+
+    // VAE decoder
+    auto vae_decoder_model = core.read_model((model_path + "/" + type + "/vae_decoder/openvino_model.xml").c_str());
+    ov::preprocess::PrePostProcessor ppp(vae_decoder_model);
+    ppp.output().model().set_layout("NCHW");
+    ppp.output().tensor().set_layout("NHWC");
+    models.vae_decoder = core.compile_model(vae_decoder_model = ppp.build(), device);
+
+    // Tokenizer
+    std::string tokenizer_model_path = "../models/tokenizer/tokenizer_encoder.xml";
```

Comment:
```suggestion
    std::string tokenizer_model_path = "../models/tokenizer/openvino_tokenizer.xml";
```
See https://github.com/openvinotoolkit/openvino_contrib/pull/792#discussion_r1431018004